src/m365/teams/commands/channel/channel-get.ts

```diff
@@ -28,17 +33,90 @@ class TeamsChannelGetCommand extends GraphCommand {
     return 'Gets information about the specific Microsoft Teams team channel';
   }
 
-  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
-    const requestOptions: any = {
-      url: `${this.resource}/v1.0/teams/${encodeURIComponent(args.options.teamId)}/channels/${encodeURIComponent(args.options.channelId)}`,
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.teamId = typeof args.options.teamId !== 'undefined';
+    telemetryProps.teamName = typeof args.options.teamName !== 'undefined';
+    telemetryProps.channelId = typeof args.options.channelId !== 'undefined';
+    telemetryProps.channelName = typeof args.options.channelName !== 'undefined';
+    return telemetryProps;
+  }
+
+  private getTeamId(args: CommandArgs): Promise<string> {
+    if (args.options.teamId) {
+      return Promise.resolve(args.options.teamId);
+    }
+
+    const teamRequestOptions: any = {
+      url: `${this.resource}/v1.0/me/joinedTeams?$filter=displayName eq '${encodeURIComponent(args.options.teamName as string)}'`,
```

Comment:
By running this endpoint you assume that the person running the query filters the Teams just by the ones he/she is member of. Administrators might want to perform the query over several Teams

To have a more general approach I would suggest that you are going to use the /v1.0/groups endpoint and then filter by resourceProvisioningOptions -> Teams